### ISSUE TITLE
Use dedicated task executor for `@ApplicationModuleListener`s

### ISSUE THREAD

**Issue by odrotbohm** [spring-projects/spring-modulith#641]
`@ApplicationModuleListener`s need dedicated control over the `Executor`  running them, for example, to enable tracing context propagation in the observability module. To contain the impact of such a customization, it makes sense to use a dedicated `Executor`.

## Implementation considerations
- Register a dedicated `ThreadPoolTaskExecutor`/`SimpleAsyncTaskExecutor` with a dedicated name and refer to that from the `@Async` annotation on `@ApplicationModuleListener` (`@ConditionalOnThreading`, `@ConditionalOnMissingBean`)

## References
- GH-637

**Comment by danstooamerican** [comment]
Will it be possible to configure the `Executor` or extend its functionality? Our application configures a custom `TaskDecorator`. Currently, we use a custom aspect to get the same functionality for `@ApplicationModuleListener`.

**Comment by odrotbohm** [comment]
Yes, you'll be able to override the bean by name. Or use a `BeanPostProcessor` to find and customize it.

**Comment by michalkopacz** [comment]
Hi team,

Hope you're doing well.

I'm just checking in on this issue regarding the dedicated task executor for @ApplicationModuleListeners. Being able to configure this executor would be very helpful for managing listener concurrency and integrating with observability tools in my application.

I was wondering if there have been any updates on the progress for this enhancement, or if it might be tentatively planned for a future release milestone?

I am also willing to contribute to implementing these changes if the team is open to external contributions in these areas.

**Comment by wimdetroyer** [comment]
Hello @odrotbohm !

I'm currently reworking a "big-ball-of-mud" application towards DDD style architecture, package-by-feature, and ofcourse extra modularity offerred by Spring Modulith. It's really been a blast so far and I'm very happy with the quality of the codebase now. Plus ofcourse all the cool things this project provides. So a big thanks first of all for your work! 👍🏻❤️


Now I'm trying to understand how observability and context propagation works, and I've stumbled upon a few issues.


---                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                           
  **(1) Overlap between this issue, #637, and Spring Boot #48033**
                                                                                                                                                                                                                                                           
  I'm wondering if this issue (or the closed #637) becomes redundant once [spring-boot#48033](https://github.com/spring-projects/spring-boot/issues/48033) ships                                           
                                                                                                                                                                                                                                                           
  From what I understand, both handle direct context propagation from the publishing thread to the async listener thread. I tested with modulith by registering a selective [`Slf4jThreadLocalAccessor`](https://github.com/micrometer-metrics/context-propagation/blob/main/context-propagation/src/main/java/io/micrometer/context/integration/Slf4jThreadLocalAccessor.java) and it works. 
                                                                                                                                                                                                                                                           
  But once Boot offers the same decorator via a property, will modulith's  conflict or will they be complementary? 

Also, because I am using modulith now, does that mean this behaviour is present everywhere, also in other parts of my application not related to modulith? 

 Does this ticket solve both these worries I have?                                                                                                                            
                                                                                                                                                                                                                                                         
  ---                                                                                                                                                                                                                                                      
   
  **(2) Context propagation on outbox replay (failed / restarted events)**                                                                                                                                                                                 
                                                                                                                                                                                                                                                         
  What I have in (1) covers the happy path (same JVM, right after commit), but what about events that fail and are replayed later  either by the republisher or after a JVM restart? It seems that context does not survive. 
                                                                                                                                                                                                                                                           
  For now I'm working around this by storing this context data (ex: `correlationId`) as fields in the event record itself. They then get serialized into the event publication table and on replay I restore them to the MDC explicitly. 

This works but it might be nice for this to be handled by the library if possible. 

For instance  **Debezium** has a dedicated [`tracingspancontext` column](https://debezium.io/documentation/reference/3.4/transformations/outbox-event-router.html#outbox-event-router-property-tracing-span-context-field) in the outbox table. I'm not an expert at all on Debezium but I suppose they then use this column to reconcile the context from within the library, instead of leaving it to the application developer.
                                              
 Would it make sense for Spring Modulith to consider something like this?
